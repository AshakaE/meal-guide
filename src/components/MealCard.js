import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../assets/css/results.module.css';

const MealCard = (props) => {
  const {
    meal: {
      id, label, mealType, dishType, cuisineType, image,
    },
  } = props;

  return (
    <div className={styles.card}>
      <div>
        <img src={image} alt={label ?? 'image'} className={styles.image} />
      </div>
      <div className={styles.cardInfo}>
        <div className={styles.mealInfo}>
          <div className={styles.label}>
            <p>Title</p>
          </div>
          <div className={styles.mealDetails}>
            <p>{label}</p>
          </div>
        </div>
        <div className={styles.mealInfo}>
          <div className={styles.mealInfoOne}>
            <div className={styles.mealTitle}>
              <p>Meal</p>
            </div>
            <div className={styles.mealTitle}>
              <p>Dish</p>
            </div>
            <div className={styles.mealTitle}>
              <p>Cuisine</p>
            </div>
          </div>
          <div className={styles.mealInfoTwo}>
            <div className={styles.labelInfo}>
              <p>{mealType}</p>
            </div>
            <div className={styles.labelInfo}>
              <p>{dishType}</p>
            </div>
            <div className={styles.labelInfo}>
              <p>{cuisineType}</p>
            </div>
          </div>
        </div>
      </div>
      <div>
        <Link to={`/recipe/${id}`}>View recipe</Link>
      </div>
    </div>
  );
};

MealCard.propTypes = {
  meal: PropTypes.shape({
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    mealType: PropTypes.arrayOf(PropTypes.string).isRequired,
    dishType: PropTypes.arrayOf(PropTypes.string).isRequired,
    cuisineType: PropTypes.arrayOf(PropTypes.string).isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
};

export default MealCard;
