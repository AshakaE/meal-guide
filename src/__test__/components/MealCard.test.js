import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MealCard from '../../components/MealCard';

describe('MealCard', () => {
  it('renders correctly', () => {
    const label = 'meal';
    const mealType = ['lunch'];
    const dishType = ['egg'];
    const cuisineType = ['test'];
    const image = 'url';
    const id = 'blank';
    const tree = renderer
      .create(
        <BrowserRouter>
          <MealCard
            meal={{
              id,
              label,
              mealType,
              dishType,
              cuisineType,
              image,
            }}
          />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
