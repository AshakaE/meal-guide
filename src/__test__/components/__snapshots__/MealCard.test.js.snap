// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCard renders correctly 1`] = `
<div
  className="card"
>
  <div>
    <img
      alt="meal"
      className="image"
      src="url"
    />
  </div>
  <div
    className="cardInfo"
  >
    <div
      className="mealInfo"
    >
      <div
        className="label"
      >
        <p>
          Title
        </p>
      </div>
      <div
        className="mealDetails"
      >
        <p>
          meal
        </p>
      </div>
    </div>
    <div
      className="mealInfo"
    >
      <div
        className="mealInfoOne"
      >
        <div
          className="mealTitle"
        >
          <p>
            Meal
          </p>
        </div>
        <div
          className="mealTitle"
        >
          <p>
            Dish
          </p>
        </div>
        <div
          className="mealTitle"
        >
          <p>
            Cuisine
          </p>
        </div>
      </div>
      <div
        className="mealInfoTwo"
      >
        <div
          className="labelInfo"
        >
          <p>
            lunch
          </p>
        </div>
        <div
          className="labelInfo"
        >
          <p>
            egg
          </p>
        </div>
        <div
          className="labelInfo"
        >
          <p>
            test
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="btn"
  >
    <a
      href="/recipe/blank"
      onClick={[Function]}
    >
      View recipe
    </a>
  </div>
</div>
`;
